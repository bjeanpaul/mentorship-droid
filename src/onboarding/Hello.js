import React, {
  PropTypes,
} from 'react';

import {
  View,
  Image,
} from 'react-native';

import { Text, Heading, Button } from 'src/components';

import { fontSize } from 'src/StyleSheet';


const Hello = ({
  name,
}) => (
  <View style={{ flex: 1 }}>
    <Image source={require('app/assets/Profile.png')} />
    <Heading style={{ fontSize: fontSize.xxLarge }}>Welcome {name}</Heading>
    <Text style={{ fontSize: fontSize.large }}>
      Help build a vibrant mentor community. Complete your profile to allow
      others learn more about you.
    </Text>
    <Button label="Complete Profile" />
  </View>
);
Hello.propTypes = {
  name: PropTypes.string,
  nextPath: PropTypes.string,
};

export default Hello;
