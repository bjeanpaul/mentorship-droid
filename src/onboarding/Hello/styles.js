import { StyleSheet } from 'react-native';

import { FONT, FONT_WEIGHT } from 'src/constants/styles';
import colors from 'src/constants/colors';


export default StyleSheet.create({
  base: {
    justifyContent: 'space-around',
  },
  imageContainer: {
    alignItems: 'center',
  },
  image: {
    height: 120,
    width: 120,
  },
  welcome: {
    fontFamily: FONT.MEDIUM,
    fontWeight: FONT_WEIGHT.MEDIUM,
    fontSize: 32,
    color: colors.PROFILE_HELLO_WELCOME_TEXT,
  },
  intro: {
    color: colors.PROFILE_HELLO_INTRO_TEXT,
    fontSize: 18,
    paddingLeft: 24,
    paddingRight: 24,
  },
});
