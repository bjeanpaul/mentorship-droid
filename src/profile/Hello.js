import React, { PropTypes } from 'react';
import { View, Image, StyleSheet } from 'react-native';

import { BaseView, Text, Button } from 'src/components';
import { font, fontWeight } from 'src/app/styles';


const IMG_PROFILE = require('app/assets/Profile.png');

const styles = StyleSheet.create({

  base: {
    justifyContent: 'space-around',
  },
  imageContainer: {
    alignItems: 'center',
  },
  image: {
    height: 120,
    width: 120,
  },
  welcome: {
    fontFamily: font.brandonTextMedium,
    fontWeight: fontWeight.medium,
    fontSize: 32,
    color: '#022428',
  },
  intro: {
    color: '#80979a',
    fontSize: 18,
    paddingLeft: 24,
    paddingRight: 24,
  },
});

const Hello = ({
  name = 'Peter',
}) => (
  <BaseView style={styles.base}>

  <View style={styles.imageContainer}>
    <Image
      style={styles.image}
      source={IMG_PROFILE}
    />
  </View>


    <Text style={styles.welcome}>Welcome {name}</Text>
    <Text style={styles.intro}>
      Help build a vibrant mentor community. Complete your profile to allow
      others learn more about you.
    </Text>
    <Button
      label="Complete Profile"
      handlePress={() => {
        // TODO: Handle transition.
      }}
    />
  </BaseView>
);
Hello.propTypes = {
  name: PropTypes.string,
};

export default Hello;
