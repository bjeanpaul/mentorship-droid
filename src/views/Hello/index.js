import React, { PropTypes } from 'react';
import { View, Image } from 'react-native';

import { BaseView, Text, Button } from 'src/components';
import images from 'src/constants/images';
import styles from './styles';


const Hello = ({
  name,
  onCompleteProfilePress,
}) => (
  <BaseView style={styles.base}>
    <View style={styles.imageContainer}>
      <Image
        style={styles.image}
        source={images.PROFILE_FACE}
      />
    </View>
    <Text style={styles.welcome}>Welcome {name}</Text>
      <Text style={styles.intro}>
      Help build a vibrant mentor community. Complete your profile to allow
      others learn more about you.
    </Text>
    <Button onPress={onCompleteProfilePress}>
      COMPLETE PROFILE
    </Button>
  </BaseView>
);
Hello.propTypes = {
  name: PropTypes.string.isRequired,
  onCompleteProfilePress: PropTypes.func.isRequired,
};

export default Hello;
