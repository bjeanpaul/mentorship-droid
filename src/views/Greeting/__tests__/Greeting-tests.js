import React from 'react';
import Hello from '../index';

describe('OnboardingStepGreeting', () => {
  function noop() {}

  it('should map props correctly', () => {
    expect(render(
      <Hello
        name="Jeff"
        onCompleteProfilePress={noop}
      />
    ).toJSON()).toMatchSnapshot();
  });

  it('should be able to tap and fire `onCompleteProfilePress`', () => {
    const onCompleteProfilePress = jest.fn();
    const el = shallow(
      <Hello
        name="Jeff"
        onCompleteProfilePress={onCompleteProfilePress}
      />
    );
    el.find('Button').simulate('press');
    expect(onCompleteProfilePress).toBeCalled();
  });
});
