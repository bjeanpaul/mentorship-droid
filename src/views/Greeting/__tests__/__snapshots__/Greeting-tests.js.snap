exports[`OnboardingStepGreeting should map props correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1
      },
      undefined
    ]
  }>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "flex": 1
        },
        Object {
          "justifyContent": "space-around"
        }
      ]
    }>
    <View
      style={
        Object {
          "alignItems": "center"
        }
      }>
      <Image
        source={1}
        style={
          Object {
            "height": 120,
            "width": 120
          }
        } />
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
            "fontSize": 13,
            "textAlign": "center"
          },
          Object {
            "color": "#022428",
            "fontFamily": "Brandon Text_medium",
            "fontSize": 32,
            "fontWeight": "400"
          }
        ]
      }>
      Welcome 
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
            "fontSize": 13,
            "textAlign": "center"
          },
          Object {
            "color": "#80979a",
            "fontSize": 18,
            "paddingLeft": 24,
            "paddingRight": 24
          }
        ]
      }>
      Help build a vibrant mentor community. Complete your profile to allow others learn more about you.
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffbcbc",
            "borderColor": "red",
            "borderWidth": 1,
            "height": 100,
            "justifyContent": "center",
            "margin": 10,
            "width": 300
          },
          undefined
        ]
      }>
      <Text
        style={
          Object {
            "color": "#333333",
            "margin": 20
          }
        }>
        TouchableNativeFeedback is not supported on this platform!
      </Text>
    </View>
  </View>
</View>
`;
